### PR TITLE
Implemented ender chests

### PR DESCRIPTION
**Pull request by T14Raptor**

### CODE REVIEW

File: server/block/ender_chest.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+func (c EnderChest) close(w *world.World, pos cube.Pos) {
+	for _, v := range w.Viewers(pos.Vec3()) {
+		v.ViewBlockAction(pos, CloseAction{})
+	}
+	w.PlaySound(pos.Vec3Centre(), sound.ChestClose{})
+}
+
+// EncodeNBT ...
+func (c EnderChest) EncodeNBT() map[string]interface{} {
+	return map[string]interface{}{}
```
Comment (Sandertv) [inline server/block/ender_chest.go:114]: This doesn't contain the `id`. Do ender chests even require block NBT at all?

File: server/player/player.go [hunk 2]
```diff
@@ -2641,10 +2651,11 @@ func (p *Player) Data() Data {
 			OffHand:      offHand,
 			MainHandSlot: p.heldSlot.Load(),
 		},
-		Effects:      p.Effects(),
-		FireTicks:    p.fireTicks.Load(),
-		FallDistance: p.fallDistance.Load(),
-		Dimension:    p.World().Dimension().EncodeDimension(),
+		EnderChestInventory: p.enderChest.Items(),
```
Comment (Sandertv) [inline server/player/player.go:2654]: I think this should use `p.enderChest.Slots()`. `p.enderChest.Items()` only returns non-zero item stacks and it doesn't retain the association of these stacks with their corresponding slot.

**Review by Sandertv — CHANGES_REQUESTED** [review]
Got two comments. Looks good otherwise.

File: server/block/ender_chest.go (hunk 1 above)
Comment (T14Raptor) [reply server/block/ender_chest.go:114]: Pretty sure they don't, but I implemented world.NBTer to retain viewer data, is there another way to do this?

Comment (Sandertv) [reply server/block/ender_chest.go:114]: That makes sense. Might want to document that so someone doesn't remove it on accident in the future.

File: server/player/player.go (hunk 2 above)
Comment (T14Raptor) [reply server/player/player.go:2654]: Indeed it should

**Review by Sandertv — APPROVED** [review]
I think this looks good. Thanks for the PR!

**Pull request merged**